### ISSUE TITLE
Allow Setting Environment Variables in a Closure

### ISSUE THREAD

**Issue by toadslop** [nushell/nushell#8520]
### Related problem

TLDR. You can't set environment variables in a close. They live in the closure's scope and then die when it ends. Export-env does not export the variables outside the closure.

### Describe the solution you'd like

I want to be able to iterate through a list and, depending on some conditions, set or hide some environment variables. Currently, it's not possible. The changes to the environment variables cannot be transferred to the parent scope.

### Describe alternatives you've considered

There isn't an alternative. If I need to set and unset environment variables based on the content of a list, I simply can't do it.

**Comment by Bahex** [comment]
This can be done by iterating over the list with a `for` loop.
All loop commands (`for`, `while`, `loop`) can modify the environment and the mutable variables of the caller scope.

If the you're using environment variables to just to have a persistent state between iterations, you can also use `reduce` or `std/iter scan` (similar to `reduce` but produces a stream rather than returning a single value).

If those do not cover your use case you can reply and I'll reopen the issue.
